src/ScopeSigs.ml

```diff
@@ -0,0 +1,118 @@
+      running the modifier [m] on it, using the internal modifier engine.
+
+      @param context The context attached to the modifier effects. *)
+
+  val modify_export : ?context:context -> hook Language.t -> unit
+  (** [modify_visible m] modifies the export namespace by
+      running the modifier [m] on it, using the internal modifier engine.
+
+      @param context The context attached to the modifier effects. *)
+
+  val modify : ?context:context -> ?prefix:Trie.bwd_path -> hook Language.t -> (data, tag) Trie.t -> (data, tag) Trie.t
+  (** Call the internal modifier engine directly on some trie. See {!val:Modifier.S.modify}.
+
+      This will not lock the current scope. *)
+
+  val export_visible : ?context:context -> hook Language.t -> unit
+  (** [export_visible m] runs the modifier [m] on the visible namespace,
+      and then merge the result into the export namespace.
+      Conflicting names during the final merge will trigger the effect [Mod.Shadowing].
+
+      @param context The context attached to the modifier effects. *)
+
+  val get_export : unit -> (data, tag) Trie.t
+  (** [get_export ()] returns the export namespace of the current scope. *)
+
+  (** {1 Sections} *)
+
+  val section : ?context_visible:context -> ?context_export:context -> Trie.path -> (unit -> 'a) -> 'a
+  (** [section p f] starts a new scope and runs the thunk [f] within the scope.
+      The child scope inherits the visible namespace from the parent, and its export namespace
+      will be prefixed with [p] and merged into both the visible and export namespaces
+      of the parent scope.
+
+      @param context_visible The context attached to the modifier effects
+      when merging the content of the section into its parent's visible namespace.
+      @param context_export The context attached to the modifier effects
+      when merging the content of the section into its parent's export namespace. *)
+
+  (** {1 Runners} *)
+
+  module Run (H : Handler with module P := P) :
+  sig
+    val run : ?export_prefix:Trie.bwd_path -> ?init_visible:(data, tag) Trie.t -> (unit -> 'a) -> 'a
+    (** [run f h] initializes a scope and executes the thunk [f], using [h] to handle modifier effects.
+
+        @param export_prefix The additional global prefix prepended to the paths reported to effect handlers
+        originating from export namespaces. The default is the empty path ([Emp]).
+        This does not affect paths originating from visible namespaces.
+        @param init_visible The initial visible namespace. The default is the empty trie. *)
+
+    val try_with : (unit -> 'a) -> 'a
+    (** Execute the code and handles the internal modifier effects. This can be used to intercept
+        or reperform those effects; for example, the following function silences the [shadow] effects.
+        See also {!val:Modifier.S.try_with}.
+
+        {[
+          let silence_shadow f = try_with f {perform with shadow = fun _ _ _ y -> y}
+        ]}
+
```

Comment:
I think this example code no longer works :laughing:

Reply:
oops! Thanks, let me fix that.